### DISCUSSION TITLE
Python 3.10.9: Several DEPRECIATION: errors when installing Glances via PIP

### DISCUSSION THREAD

**Discussion by mkanet** [nicolargo/glances#2305] [Q&A]
I executed the command `pip install glances[all]` on a virgin Python 3.10.9 installation on a clean Win11 (64bit) PC.  

I am getting all the all the **DEPRECATION:** errors below.  **I would like to know what is the cleanest and most minimal way to mitigate the errors below**...  I don't want to install any unnecessary pieces of software/modules/files.

Also, I would also like to know how to fix the error: **Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools":** error.  I downloaded `vs_BuildTool.exe` installer, however, its not clear what component I need to select.  If there is a way to install ONLY the missing component, please let me know how to do it.
**NOTE:** Can I Select **_Workloads_**, then select **_Windows 11 SDK (10.0.22000.0)_** and **_MSVC v143 - VS 2022 C++ x64/x86 build tools (latest)_** per these instructions https://www.scivision.dev/python-windows-visual-c-14-required?

.
**ERRORS**
```
 DEPRECATION: scandir is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for scandir ... done

 DEPRECATION: pbkdf2 is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for pbkdf2 ... done

 DEPRECATION: netifaces is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559

  Running setup.py install for netifaces ... error
  error: subprocess-exited-with-error

  × Running setup.py install for netifaces did not run successfully.
  │ exit code: 1
  ╰─> [9 lines of output]
      C:\Users\Michael\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\config\setupcfg.py:508: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      running install
      C:\Users\Michael\AppData\Local\Programs\Python\Python310\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_ext
      building 'netifaces' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> netifaces

note: This is an issue with the package mentioned above, not pip.
hint: See above for output from the failure.
```

#### Versions

* Glances & psutil (glances -V):   Trying to install the latest version via PIP
* Operating System (lsb_release -a):  Windows 11 64bit
* How do you install Glances (Pypi package, script, package manager, source):  PIP
![errors](https://user-images.githubusercontent.com/811008/227058982-559146a4-23c8-4a60-bb8f-c5832f73b8e9.png)

**Comment by RazCrimson** [comment]
@mkanet 
The issue is due to the fact `netifaces` is currently unmaintained, so there are no compiled windows binary matching py310 to install. Pip then tries to manually compile it which requires `Microsoft C++ Build Tools**`.

**Option 1**
If you don't really plan to use all plugins (which is usually the case), then the simplest way would be to start with `pip install glances`, which would install the just the core plugins. You can then try to install dependencies of just the plugins u would need.

**Option 2**
If you really want all the plugins, then you need to go download the Visual Studio Installer, and just install the  `Microsoft C++ Build Tools` ver 14.0 or later using it as the error message says. 

Personal Opinion: It's usually big pain to get Build Tools working right.

**Reply by mkanet** [reply]
@RazCrimson  thank you so much helping.  Up until about a year ago, I used to be able to do:  `pip install glances[all]` without getting the error about netifaces on another PC. 

I think I fould the latest netifaces whl file here:
https://www.lfd.uci.edu/~gohlke/pythonlibs/#netifaces

Can't I just use **_netifaces-0.11.0-cp39-cp39-win_amd64.whl_** from https://www.lfd.uci.edu/~gohlke/pythonlibs/#netifaces ?

Then, execute:  `pip install C:\Patht\to\netifaces-0.11.0-cp39-cp39-win_amd64.whl`?  I'm not sure if its compatible with **Python 3.10.9**

I would really appreciate your guidance.  I'm just not sure if thats the best version of netifaces to use.  **Also, when I installed Glances on my other PC, I didn't get that error, in fact, it downloaded: `netifaces-0.11.0.tar.gz` and installed it.  It didn't have to build anything.**

**Reply by RazCrimson** [reply]
If you are willing to trust the website and install the compiled binaries from there, then it's worth a shot. It might work with py310. In case it doesn't work, you can try to downgrade your python version to py39 and do the install again.

The commands to install that you have listed should work fine.

 > I would really appreciate your guidance. I'm just not sure if thats the best version of netifaces to use. Also, when I installed Glances on my other PC, I didn't get that error, in fact, it downloaded: netifaces-0.11.0.tar.gz and installed it. It didn't have to build anything.

Maybe it has build tools installed or you had an older py version that has windows binaries on pypi.

**Reply by mkanet** [reply]
@RazCrimson I ended up installing MS BuildTools on another PC that has the same Python version... and installed Glances[all] on that PC, then manually copied the below directory with its contents and respective compiled binary to the new PC... Then, immediately I was able to import the module and use it in my python scripts.

```
C:\Users\Michael\AppData\Local\Programs\Python\Python310\Lib\site-packages\netifaces-0.11.0-py3.10.egg-info\*

C:\Users\Michael\AppData\Local\Programs\Python\Python310\Lib\site-packages\netifaces.cp310-win_amd64.pyd
```

**Reply by RazCrimson** [reply]
Cool, that should not cause problems.
